### ISSUE TITLE
Extend aiders --yes option

### ISSUE THREAD

**Issue by p-wegner** [paul-gauthier/aider#1375]
### Issue

I'm currently writing a Jetbrains IDE Integration Plugin that runs aider in scripting mode and having more control over the different confirm use cases for the` --yes` option would be great. 

`--no-suggest-shell-commands `helps with this, but if i only want to allow file creation and no changes to the context, i can't prevent this, can i? 
For my use case something like `--no-context-changes` and/or `--prevent-file-creation` would be helpful. I would usually allow file creation, but no context changes/web scraping in my use case.

### Version and model info

Aider v0.54.7

**Comment by p-wegner** [comment]
Additionally, using --yes will also automatically update aider or install needed packages like aider[help] if needed, which can take quite some time (torch + indexing). Especially when used for scripting and relying on the API of a fixed aider version, this is rather inconvenient.

**Comment by paul-gauthier** [comment]
You should consider scripting aider from python to get more control.

**Comment by p-wegner** [comment]
Yeah, that would solve those constraints. As of now, keeping requirements on the env minimal (which e.g. python version) is the goal. Devs working with any Jetbrains IDE should be able to simply install https://plugins.jetbrains.com/plugin/25249-coding-aider without having to care about which python version is available on the system. Currently a compatible Aider has to be present of course, but i'll likely add an option to run it with docker https://github.com/p-wegner/coding-aider/discussions/7

**Comment by sutt** [comment]
I've created a fork of v0.85.5 that does YOLO here: https://github.com/sutt/agent-aider
